### DISCUSSION TITLE
x.user extension auto-load does not work

### DISCUSSION THREAD

**Discussion by zetazeroalfa** [mecha-cms/mecha#267] [Q&A] [answered]
Hi, I need little help, my mecha-cms site does not detect user and panel extensions installation...after I upload them in x folder I back to navigate web site but I'm not seeing execution of UI to create new user.
What can be my problem?

**Comment by taufik-nurrohman** [comment]
Do `.\lot\x\user\index.php` and `.\lot\x\panel\index.php` files exist in your installation?

> After I upload them in x folder

I am assuming you are uploading them manually.

**Comment by zetazeroalfa** [comment]
Yes, I uploaded them manually. Something wrong?
Each extension has its original index.php file as it come in its version
I'm using:
- x.user-main.zip from github.com/mecha-cms/x.user
- x.panel@v2.8.1.zip from store https://mecha-cms.com

**Reply by taufik-nurrohman** [reply]
The panel version is outdated, and the user version is unstable. Either try to install third of them from the `main` branch or use the [installer file](https://github.com/mecha-cms/start).

**Note:** `x.user-main.zip` must have a single folder level named `x.user-main` in it which should not be there. The `index.php` file must be in the root.

I am currently developing the new site design at https://dev.mecha-cms.com you may want to try to download the files from there instead.

**Reply by zetazeroalfa** [reply]
I uploaded the version you suggested from dev.mecha-cms.com (x.user@v2.0.0.zip)...still nothing! it doesn't work anyway...

My extension directory is .\lot\x\user (containing its index.php too)

It seems the user extension is completely ignored... I don't understand how cms sholud be detect it... can you confirm if I have to navigate cms home or launch another page to get user login form? and is there some way to debug and catch the call to the extension?

Thank you in advance.

**Reply by taufik-nurrohman** [reply]
**Option 1:** Try to navigate to the `/user` path.

**Option 2:** Delete the `.\lot\user` folder with its contents and see if you are redirected to the user registration page after doing so.

To check if there are some errors, first, make sure the test mode is active. That can be done by changing the `TEST` constant value to `true`. At this time, the value should be `true` by default because version 3.0.0 is not yet ready for production.

Visit your site again and then check if there is an `error`, `error-x`, and/or `error-y` file exists in `.\engine\log` folder.

**Reply by zetazeroalfa** [reply]
1) navigating `/user` path I get this content output: `ಠ_ಠ`
2) after deleted `.\lot\user` and nothing changes  

I don't understand what version 3.0.0 you refer ...as you recommended now i am working with version 2.0.0 of user extension (on mecha v.2.6.4). Anyway i added TEST constant in my root `.\index.php` as following:

`define('TEST',true);`

I keep seeing error code `ಠ_ಠ` and folder `.\engine\log` is still empty.

**Reply by taufik-nurrohman** [reply, marked as answer]
> on mecha v.2.6.4

Ah, that&rsquo;s the problem. For now, please download the core version from [the development site](https://dev.mecha-cms.com), which is curently on version 3.0.0.

![Screenshot 2023-06-06 at 00-44-22 Home · Mecha CMS](https://github.com/mecha-cms/mecha/assets/1669261/7888ee80-4ec0-46c9-b481-c8fa7e26dc3d)

**Reply by taufik-nurrohman** [reply]
If you insist on using the version 2.6, you will need to use the older versions which can be obtained via the release page. User extension version 1.13.0 is probably enough.

https://github.com/mecha-cms/x.user/releases

**Reply by taufik-nurrohman** [reply]
The new site design has been moved to the root domain. I will try to get rid of the confusions as quick as I can.

**Reply by zetazeroalfa** [reply]
After upgrade extensions seem work fine. 
So for now where is it possible configure layout files (before, after, header footer,etc)?

**Reply by taufik-nurrohman** [reply]
Layout files are stored in the `.\lot\y\*` folder. The active layout is the one that has an `index.php` file. Learn more at https://mecha-cms.com/reference/3.x/layout/1.

**Reply by zetazeroalfa** [reply]
I'm not talking about layout for backend panel, I'm talking about public web site base layout... I had customized web site theme with php files in /lot/x/layout/* and now, after upgrade, graphic and theme resources are no more loaded. 
Is something changed about theme management in the last version?

**Reply by taufik-nurrohman** [reply]
Yes. Themes no longer stored in `.\lot\layout` now they are stored in `.\lot\y\theme-name`.

See the reasons behind the changes here: https://github.com/orgs/mecha-cms/discussions/205
